src/main/java/parser/Parser.java

```diff
@@ -0,0 +1,244 @@
+				int min = Integer.parseInt(dateTimeArr[1].substring(2, 4));
+				time = LocalTime.of(hr, min);
+			} catch (NumberFormatException ex) {
+				ui.sendResponse(dialogContainer, storage, ui.createLabel(errorMessage));
+				return null;
+			}
+		}
+
+		try {
+			if (dateTimeStr.contains("/")) { // Input Format: dd/MM/YYYY
+				dateArr = Stream.of(dateTimeArr[0].split("/"))
+						.map(Integer::valueOf)
+						.toArray(Integer[]::new);
+				return LocalDateTime.of(LocalDate.of(dateArr[2], dateArr[1], dateArr[0]), time);
+			}
+
+			// Input Format: YYYY-MM-dd
+			dateArr = Stream.of(dateTimeArr[0].split("-"))
+					.map(Integer::valueOf)
+					.toArray(Integer[]::new);
+			return LocalDateTime.of(LocalDate.of(dateArr[0], dateArr[1], dateArr[2]), time);
+		} catch (ArrayIndexOutOfBoundsException ex) {
+			ui.sendResponse(dialogContainer, storage, ui.createLabel(errorMessage));
+			return null;
+		}
+
+	}
+
+	/**
+	 * Get a datetime object from deadline tasks.
+	 * 
+	 * @param inputArr
+	 * @param ui
+	 * @param storage
+	 * @param dialogContainer
+	 * @return LocalDateTime
+	 */
+	public LocalDateTime getBy(String[] inputArr, Ui ui, Storage storage, VBox dialogContainer) {
+		int l = inputArr.length;
+		for (int i = 0; i < l; i++) {
+			if (i < l - 1 && inputArr[i].equals("/by")) {
+				return parseDateTimeStr(sliceArrAndConcate(inputArr, i + 1, l), ui, storage, dialogContainer);
+			}
+		}
+		String errorMessage = "Oops! The description of a DEADLINE must include a BY datetime";
+		ui.sendResponse(dialogContainer, storage, ui.createLabel(errorMessage));
+		return null;
+	}
+
+	/**
+	 * Get a datetime array object from event tasks.
+	 * 
+	 * @param inputArr
+	 * @param ui
+	 * @param storage
+	 * @param dialogContainer
+	 * @return LocalDateTime[]
+	 */
+	public LocalDateTime[] getFromTo(String[] inputArr, Ui ui, Storage storage, VBox dialogContainer) {
+		int l = inputArr.length;
+		int fromStartIdx = -1, fromEndIdx = l, toStartIdx = -1;
+		LocalDateTime from = null;
+		LocalDateTime to = null;
+		for (int i = 0; i < l; i++) {
+			if (i < l - 1) {
+				if (inputArr[i].equals("/from")) {
+					fromStartIdx = i + 1;
+				}
+				if (inputArr[i].equals("/to")) {
+					fromEndIdx = i;
+					toStartIdx = i + 1;
+				}
+			}
+		}
+		if (fromStartIdx > -1) {
+			from = parseDateTimeStr(sliceArrAndConcate(inputArr, fromStartIdx, fromEndIdx), ui, storage,
+					dialogContainer);
+			if (from != null && toStartIdx > -1) {
+				to = parseDateTimeStr(sliceArrAndConcate(inputArr, toStartIdx, l), ui, storage, dialogContainer);
+			}
+		}
+		return new LocalDateTime[] { from, to };
+	}
+
+	/**
+	 * Remove the datetime string from an input string.
+	 * 
+	 * @param input
+	 * @return String
+	 */
+	public String trimDate(String input) {
+		String[] inputArr = input.split(" ");
+		StringBuilder sb = new StringBuilder();
+		for (String s : inputArr) {
+			if (s.contains("/")) {
+				return sb.toString().trim();
+			}
+			sb.append(s + " ");

```

Comment:
I believe a more meaningful variable name for each String could be used instead of `s`, such as `untrimmedDateStr`